File: frontend/src/types/room.ts
export interface RoomStatus {
  sala_id: string
  ocupada: boolean
  luminosidade: boolean
  ultimo_movimento: string
  tempo_vazia: number
  uptime?: number
  rssi?: number
  ip?: string
}

export interface RoomHistory {
  timestamp: string
  movimento: number
  luminosidade?: number
}

export type Comando = 'ON' | 'OFF'

export type LogTipo = 'movimento' | 'luminosidade' | 'remoto' | 'alerta' | 'sistema'

export interface LogEntry {
  timestamp: string
  mensagem: string
  tipo?: LogTipo
}

File: backend/src/models/room.py
from pydantic import BaseModel
from datetime import datetime


class RoomStatus(BaseModel):
    sala_id: str
    ocupada: bool
    luminosidade: bool
    ultimo_movimento: datetime
    tempo_vazia: int  # segundos desde o último movimento
    uptime: int | None = None   # segundos desde o boot do ESP32
    rssi: int | None = None     # sinal Wi-Fi em dBm
    ip: str | None = None       # IP do ESP32 na rede


class RoomHistory(BaseModel):
    timestamp: datetime
    movimento: int
    luminosidade: int | None = None


class CommandPayload(BaseModel):
    comando: str  # "ON" | "OFF"
